File: BEE 1047.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    int h1, m1, h2, m2, total_hour, total_minute;
    cin >> h1 >> m1 >> h2 >> m2;
    if ((h1 == h2) && (m1 == m2)){
        total_hour = (24 - h1) + h2;
        total_minute = m2 - m1; 
    }
    else if ((h1 == h2) && (m1 < m2)){
        total_hour = h2 - h1;
        total_minute = m2 - m1; 
    }
    else if ((h1 == h2) && (m1 > m2)){
        total_hour = (24 - h1) + h2 - 1;
        total_minute = (60 - m1) + m2; 
    }
    else if ((h1 < h2) && (m1 == m2)){
        total_hour = h2 - h1;
        total_minute = m2 - m1;
    }
    else if ((h1 > h2) && (m1 == m2)){
        total_hour = (24 - h1) + h2;
        total_minute = m2 - m1;
    }
    else if ((h1 < h2) && (m1 < m2)){
        total_hour = h2 - h1;
        total_minute = m2 - m1;
    }
    else if ((h1 < h2) && (m1 > m2)){
        total_hour = h2 - h1 - 1;
        total_minute = (60 - m1) + m2;
    }
    else if ((h1 > h2) && (m1 < m2)){
        total_hour = (24 - h1) + h2 - 1;
        total_minute = m2 - m1;
    }
    else{
        total_hour = (24 - h1) + h2 - 1;
        total_minute = (60 - m1) + m2;
    }
    cout << "O JOGO DUROU " << total_hour << " HORA(S) E " << total_minute << " MINUTO(S)" << endl;
    return 0;
}

File: BEE 1047.php
<?php

    list($initial_hour, $initial_minute, $final_hour, $final_minute) = explode(" ", readline());
    
    $total_minute = 0;
    $total_hour = 0;
    
    if($initial_hour == $final_hour){
        if ($initial_minute == $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = (24 - $initial_hour) + $final_hour;
        }
        elseif ($initial_minute < $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = $final_hour - $initial_hour;
        }
        else{
            $total_minute = (60 - $initial_minute) + $final_minute;
            $total_hour = (24 - $initial_hour) + $final_hour - 1;
        }
    }
    elseif($initial_hour < $final_hour){
        if ($initial_minute == $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = $final_hour - $initial_hour;
        }
        elseif ($initial_minute < $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = $final_hour - $initial_hour;
        }
        else{
            $total_minute = (60 - $initial_minute) + $final_minute;
            $total_hour = $final_hour - $initial_hour - 1;
        }
    }
    else{
        if ($initial_minute == $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = (24 - $initial_hour) + $final_hour;
        }
        elseif ($initial_minute < $final_minute){
            $total_minute = $final_minute - $initial_minute;
            $total_hour = (24 - $initial_hour) + $final_hour - 1;
        }
        else{
            $total_minute = (60 - $initial_minute) + $final_minute;
            $total_hour = (24 - $initial_hour) + $final_hour - 1;
        }
    }

    echo "O JOGO DUROU {$total_hour} HORA(S) E {$total_minute} MINUTO(S)\n";

?>